### DISCUSSION TITLE
[Question] How to change the friction of the rigidobject from the Usd or urdf ?

### DISCUSSION THREAD

**Discussion by yichaoshen-MS** [isaac-sim/IsaacLab#1578] [Q&A]
### Question

Unlike the object from the "sim_utils.**CuboidCfg**", it cannot set the **friction** of "**physics_material**" in the "sim_utils.**UsdFileCfg**"?

It seems only can set global **friction** by spawning physics_material in "SimulationCfg"?

**Comment by Mayankm96** [comment]
My recommendation is to do it through PhysX directly (at runtime) as having this information at the USD-level gets slightly complicated due to the instancing of meshes. We have the event term for randomizing PhysX material here: https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/events.py#L34

**Reply by YannickP99** [reply]
Hey @Mayankm96 ,
is there a similar function to set the material properties for a RigidBodyCollection? Or can I somehow apply the function on each object of the collection? I define the collection as following:
```
models = RigidObjectCollectionCfg(rigid_objects={})

# Read in original CAD dimensions
with open(f"{DATA_FOLDER}/{OBJECT_NAME}.yaml", 'r') as file:
object_data = yaml.safe_load(file)

colors = generate_colors(num_models)
for index in range(0,num_models):

	color = (1/(num_models-1), 1-1/(num_models-1), 0.0)
	height = round(model_height_min + index * (model_height_max - model_height_min)/(num_models-1),2)
	scaling = height/object_data["height"]

	model = RigidObjectCfg(
	    prim_path=f"/World/envs/env_.*/model_{index}",
	    init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, 0.0), rot=(0.7071068, 0.7071068, 0, 0)),
	    spawn=sim_utils.UsdFileCfg(
		usd_path=f"{USD_FOLDER}/{OBJECT_NAME}.usd",
		scale = (scaling, scaling, scaling),
		mass_props=sim_utils.MassPropertiesCfg(mass=1.0),
		rigid_props=sim_utils.RigidBodyPropertiesCfg(
		                                    max_depenetration_velocity=1.0, 
		                                    disable_gravity=False,
		                                    kinematic_enabled=True,    # fix model position
		),
		collision_props=sim_utils.schemas.CollisionPropertiesCfg(collision_enabled=True),
		visual_material=sim_utils.PreviewSurfaceCfg(diffuse_color=colors[index]),
	    )
	)

models.rigid_objects["model_" + str(index)] = model 
```

Also an additional question: is there some information about the friction in Isaac Sim/Lab? Because I am wondering what physical meaning of the values is as they are different from friction coefficients.

**Reply by imendezval** [reply]
Hey! did you ever find a solution for this?

**Reply by YannickP99** [reply]
Hey @imendezval, unfortunately I did not
